geps/gep-2162.md

```diff
@@ -0,0 +1,112 @@
+# GEP-2162: Supported features in GatewayClass Status
+
+* Issue: [#2162](https://github.com/kubernetes-sigs/gateway-api/issues/2162)
+* Status: Provisional
+
+## TLDR
+
+This GEP proposes to enhance the [GatewayClassStatus](https://github.com/kubernetes-sigs/gateway-api/blob/f2cd9bb92b4ff392416c40d6148ff7f76b30e649/apis/v1beta1/gatewayclass_types.go#L185) to include a list of Gateway API features supported by the installed GatewayClass. 
+
+## Goals
+
+* Improve UX by enabling users to easily see what features the implementation(GatewayClass) support.
+
+* Automatically run conformance tests based on the supported features populated in GatewayClass status
+
+## Non-Goals
+
+* Validate correctness of supported features published by the implementation. Meaning we don't intend to verify whether the supported features reported by the implementation are indeed supported.
+
+## Introduction
+
+The current [GatewayClassStatus](https://github.com/kubernetes-sigs/gateway-api/blob/f2cd9bb92b4ff392416c40d6148ff7f76b30e649/apis/v1beta1/gatewayclass_types.go#L185 is only used to store conditions the controller publishes.
+
+Partnered with the [Conformance Profiles](https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-1709.md) work, we want to:
+
+1. Improve UX by enabling users to easily see what features the implementation(GatewayClass) support.
+1. Automatically run conformance tests based on the supported features populated in GatewayClass status.
+1. Potentially provide clear errors when a user is trying to use an unsupported feature (more under [Future Work](#future-work)).
+
+This doc proposes to enhance the GatewayClassStatus API so implementations could publish a list of features they support/don't support.
+
+Implementations **should** publish the supported/unsupported features before Accepting the GatewayClass.
+
+They can either maintain static lists of supported/unsupported features OR generate them from their conformance tests results/reports as part of their CI.
```

Comment:
```suggestion
Implementations are free to decide how they manage this information - common approaches could be either to maintain static lists of supported/unsupported features OR generate them from their conformance tests results/reports as part of their CI.
```

We don't want to imply that people can't come up with some other way to do this, all we really care about is the end result (that is, that the information is in the status).